ReactiveCocoa/Swift/Signal.swift

```diff
@@ -410,16 +410,126 @@ public func takeUntil<T, E>(trigger: Signal<(), NoError>)(signal: Signal<T, E>)
 	}
 }
 
+/// Forwards events from `signal` with history: values of the returned signal
+/// are a tuple whose first member is the previous value and whose second member
+/// is the current value. `initial` is supplied as the first member when `signal`
+/// sends its first value.
+public func combinePrevious<T, E>(initial: T)(signal: Signal<T, E>) -> Signal<(T, T), E> {
+	return signal |> scan((initial, initial)) { previousCombinedValues, newValue in
+		return (previousCombinedValues.1, newValue)
+	}
+}
+
+/// Like `scan`, but sends only the final value and then immediately completes.
+/// If `signal` completes before you observe the resulting signal, the resulting
+/// signal may send no values before completing.
```

Comment:
The resulting signal wouldn't send anything in that case, would it? This is a more general problem with `Signal`s, because you can miss events. (And so this documentation would apply to every operator.)

This is why [I don't want to support](https://github.com/ReactiveCocoa/ReactiveCocoa/pull/1689#issuecomment-70707397) operators like `concat` over hot signals.